tests/handlers/test_saml.py

```diff
@@ -19,6 +19,24 @@
 
 from tests.unittest import HomeserverTestCase, override_config
 
+# Check if we have the tests.
+try:
+    import saml2.config
+    from saml2.sigver import SigverError
+
+    has_saml2 = True
+
+    # pysaml2 can be installed and imported, but might not be able to find xmlsec1.
+    config = saml2.config.SPConfig()
+    try:
+        config.load({"metadata": {}})
+        has_xmlsec1 = True
+    except SigverError:
+        has_xmlsec1 = False
```

Comment:
I think we could call `get_xmlsec_binary` manually if we wanted, but since this is the code-path we use (in `saml_config` it seemed nicer to do this?

Reply:
Assuming this covers all exception cases, it looks quite nice to me.

Reply:
I didn't notice anything else that could cause an issue, but no promises. 😄